### ISSUE TITLE
TypeError: Cannot read property 'name' of null at Object.Module._extensions..js

### ISSUE THREAD

**Issue by timoxley** [standard-things/esm#434]
Just logging this here for searchability for anyone else getting this specific error:

```
TypeError: Cannot read property 'name' of null
    at Object.Module._extensions..js (module.js:663:10)
```

### Check that your default exports are using named functions.

i.e.

```js
// bad
export default function () {

}

// good
export default function MyName () {

}
```
Obviously you should be able to have anonymous default exports, but some tooling seems to be tripped up by it – **what I'm proposing above is just a workaround**.

**Note:** this is not a solution for just any generic "Cannot read property X of Y", this is specifically for the error trying to read "name" and "null" when using `esm` in node.

----

**Could not find a minimal repro case**, anonymous default exports seem to work just fine in simple cases, but I'm using a monorepo with symlinked dependencies, perhaps it's something to do with that? 

This just started happening after doing a clean install without any lockfiles, so it appears that something changed in some sub-dependency of `esm`, maybe? It wasn't clear why from the stack trace or even from stepping through the code with a debugger what the issue was. **Maybe it's not even `esm` related**, maybe it's something else entirely, I don't have the time to track it down. Got tipped off by this issue, however as far as I know I'm not using any esdoc anywhere: https://github.com/esdoc/esdoc/issues/11

Feel free to close this issue.

**Comment by jdalton** [comment]
Hi @timoxley!

This shouldn't be erroring for sure. If you produce a minimum repro that would rock. What version of Node and `esm` are you using?

**Comment by jdalton** [comment]
You can use the `esm` option `"debug":true` to possibly get a better stack trace too.

**Comment by jdalton** [comment]
[v3.0.42](https://github.com/standard-things/esm/releases/tag/3.0.42) is released!

**Comment by cinnabarcaracal** [comment]
I've also just had this (or something very similar) start happening in a medium sized CRA react project.

I have `"esm": "^3.2.25"` as a devDependency for some dev helper scripts that live in the same repo but aren't built with the main project. I started getting this error after I cleared my node_modules, removed my lock file and did a new yarn install. Naming the default function export in a couple of my reducers got my project to build again.

```
TypeError: Cannot read property 'name' of null
    at Array.filter (<anonymous>)
```
